src/app/store/defects/selectors/defects.selectors.ts

```diff
@@ -63,6 +65,6 @@ export const hgvDefects = filteredDefects(VehicleTypes.HGV);
 
 export const trlDefects = filteredDefects(VehicleTypes.TRL);
 
-export const defect = (id: string) => createSelector(defectsFeatureState, state => state.entities[id]);
+export const defect = (id: string) => createSelector(defectsFeatureState, (state) => state.entities[`${id}`]);
```

Comment:
curious what people think of this syntax, it stops the warning for the Generic Object Injection Sink, there are quite a few of them and I didn't want to just ignore them

Reply:
Been using that too. template literal makes whatever expression is in that id into a string